### DISCUSSION TITLE
Help adding unsupported devices

### DISCUSSION THREAD

**Discussion by cherowley** [Koenkk/zigbee2mqtt#6154] [Devices]
Hi!

Are there anymore places for help besides the official docs? I'm stuck on what to put in the fromzigbee.js file.

I've done the devices.js (not that the exposes item is correct in it:

    //KAMI
    {
    zigbeeModel: ['Z3ContactSensor'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'kami entry sensor', // Vendor model number, look on the device for a model number
    vendor: 'Silicon Labs', // Vendor of the device (only used for documentation and startup logging)
    description: 'Z3 Contact sensor ', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    supports: 'Contact', // Actions this device supports (only used for documentation)
    fromZigbee: [],
    toZigbee: [],
    exposes: [e.contact(), e.battery_low()],
    }

Debug logs for a pir and door contact:

PIR:

Debug Received Zigbee message from 'PIR', type 'raw', cluster 'msOccupancySensing', data '{"data":[3,0,2,0,0,0,24,1],"type":"Buffer"}' from endpoint 3 with groupID 0
Debug No converter available for 'kami entry sensor' with cluster 'msOccupancySensing' and type 'raw' and data '{"data":[3,0,2,0,0,0,24,1],"type":"Buffer"}'

Door open:

Debug Received Zigbee message from 'Contact 1', type 'raw', cluster 'ssIasZone', data '{"data":[3,0,2,0,0,0,48,1],"type":"Buffer"}' from endpoint 2 with groupID 0
Debug No converter available for 'kami entry sensor' with cluster 'ssIasZone' and type 'raw' and data '{"data":[3,0,2,0,0,0,48,1],"type":"Buffer"}'

Door Closed:
Debug Received Zigbee message from 'Contact 1', type 'raw', cluster 'ssIasZone', data '{"data":[3,0,2,0,0,0,48,0],"type":"Buffer"}' from endpoint 2 with groupID 0
Debug No converter available for 'kami entry sensor' with cluster 'ssIasZone' and type 'raw' and data '{"data":[3,0,2,0,0,0,48,0],"type":"Buffer"}'

For some reason I can't figure out how to translate and what to put, sorry for being dense :(

**Comment by oscarliuyc** [comment]
Hi,

Obviously, the key is to read the list data element in the array, 1 = door open, = door close.  Actually, I'm stuck in a similar situation when writing a converter for this data structure.

**pseudo code**

cluster == 'sslasZone';
type == 'raw'
if msg.data.data[8] = 1, return occupancy= door closed
if msg.data.data[8] = 0, return occupancy= door open

seems that syntax "msg.data.data[8] " is wrong.  Hope somebody can help.

**Comment by cherowley** [comment]
Thanks, yes there's quite a bit of info missing on what to code how to handle the raw message...

**Reply by cherowley** [reply]
Has anyone got any pointers to tutorials/help (other than the main one as that is not in-depth enough for me) out there?

Cheers!

**Comment by cherowley** [comment]
Ok, so no other resources found by google, so any experts on here willing to help? Seems a shame to ditch this software just because of this small sticking point :)

**Comment by Koenkk** [comment]
Seems this sensor sends non standard zigbee messages (because of `raw`). With the following changes contact should work: https://github.com/Koenkk/zigbee-herdsman-converters/pull/2247/files

**Reply by cherowley** [reply]
Fab! Thankyou Koenkk!

I altered one bit 

    return {contact: msg.data.data[7] === 0};

to

    return {contact: msg.data[7] === 0};


How would I go about allowing though both

  cluster 'msOccupancySensing'

and

 cluster 'ssIasZone'

As the PIR sensor returns the same raw data but just under the different cluster...

 No converter available for 'Z3ContactSensor' with cluster 'msOccupancySensing' and type 'raw' and data '{"data":[3,0,2,0,0,0,24,1],"type":"Buffer"}'

(i tried duplicating the code in fromzigbee.js changing to msOccupancySensing but it only seems to allow one and ignore the other...

**Reply by Koenkk** [reply]
Updated https://github.com/Koenkk/zigbee-herdsman-converters/pull/2247/files with occupancy

**Reply by cherowley** [reply]
Ah I see! Many thanks all good now :)
